x-pack/plugin/text-structure/src/test/java/org/elasticsearch/xpack/textstructure/structurefinder/LogTextStructureFinderTests.java

```diff
@@ -99,6 +99,120 @@ public void testCreateConfigsGivenElasticsearchLog() throws Exception {
+            assertNull(structure.getJodaTimestampFormats());
+            FieldStats messageFieldStats = structure.getFieldStats().get("message");
+            assertNotNull(messageFieldStats);
+            for (String statMessage : messageFieldStats.getTopHits()
+                .stream()
+                .map(m -> (String) m.get("value"))
+                .collect(Collectors.toList())) {
+                assertThat(structureFinder.getSampleMessages(), hasItem(statMessage));
+            }
+            assertEquals(Collections.singleton("properties"), structure.getMappings().keySet());
+        }
+
+        {
+            // Finally, test the behaviour when parsing single-line log messages that _do_ contain timestamps
+            // but the timestamp format override has been set to TextStructureUtils.NULL_TIMESTAMP_FORMAT ("null")
+            TextStructureFinder structureFinder = factory.createFromSample(
+                explanation,
+                TEXT_SAMPLE,
+                charset,
+                hasByteOrderMarker,
+                TextStructureFinderManager.DEFAULT_LINE_MERGE_SIZE_LIMIT,
+                overrides,
+                NOOP_TIMEOUT_CHECKER
+            );
+
+            TextStructure structure = structureFinder.getStructure();
+
+            assertEquals(TextStructure.Format.SEMI_STRUCTURED_TEXT, structure.getFormat());
+            assertEquals(charset, structure.getCharset());
+            if (hasByteOrderMarker == null) {
+                assertNull(structure.getHasByteOrderMarker());
+            } else {
+                assertEquals(hasByteOrderMarker, structure.getHasByteOrderMarker());
+            }
+            assertNull(structure.getExcludeLinesPattern());
+            assertNull(structure.getMultilineStartPattern());
+            assertNull(structure.getDelimiter());
+            assertNull(structure.getQuote());
+            assertNull(structure.getHasHeaderRow());
+            assertNull(structure.getShouldTrimFields());
+            // a timestamp field is detected but it's not set to be the primary one.
+            assertEquals(
+                "\\[%{TIMESTAMP_ISO8601:extra_timestamp}\\]"
+                    + "\\[%{LOGLEVEL:loglevel} \\]\\[.*?    .*? .*? .*? .*? .*? .*? .*? .*? .*? .*?\\].*?",
+                structure.getGrokPattern()
+            );
+            assertNull(structure.getTimestampField());
+            assertNull(structure.getJodaTimestampFormats());
+            FieldStats messageFieldStats = structure.getFieldStats().get("message");
+            assertNotNull(messageFieldStats);
+            for (String statMessage : messageFieldStats.getTopHits()
+                .stream()
+                .map(m -> (String) m.get("value"))
+                .collect(Collectors.toList())) {
+                assertThat(structureFinder.getSampleMessages(), hasItem(statMessage));
+            }
+            assertEquals(Collections.singleton("properties"), structure.getMappings().keySet());
```

Comment:
This isn't a very important assertion. What's much more interesting is what's in the level _under_ `properties`. It should not contain `@timestamp`.